frontend/src/concepts/pipelines/content/createRun/contentSections/ParamsSection/ParamsSection.tsx

```diff
@@ -0,0 +1,107 @@
+import * as React from 'react';
+import { Alert, FormGroup, FormSection, TextInput } from '@patternfly/react-core';
+import {
+  CreateRunPageSections,
+  runPageSectionTitles,
+} from '~/concepts/pipelines/content/createRun/const';
+import {
+  InputDefParamType,
+  PipelineVersionKFv2,
+  RuntimeConfigParameters,
+} from '~/concepts/pipelines/kfTypes';
+import { getInputDefinitionParams } from '~/concepts/pipelines/content/createRun/utils';
+import NumberInputWrapper from '~/components/NumberInputWrapper';
+import { RadioInputParam } from './RadioInputParam';
+import { JsonInputParam } from './JsonInputParam';
+
+type ParamsSectionProps = {
+  runParams: RuntimeConfigParameters | undefined;
+  version: PipelineVersionKFv2 | null;
+  onChange: (params: RuntimeConfigParameters) => void;
+};
+
+export const ParamsSection: React.FC<ParamsSectionProps> = ({
+  runParams = {},
+  version,
+  onChange,
+}) => {
+  const renderContent = (): React.ReactNode => {
+    if (!version) {
+      return (
+        <Alert
+          variant="info"
+          isInline
+          isPlain
+          title="You must select a pipeline and version before you can set parameters."
+        />
+      );
+    }
+
+    if (!Object.keys(runParams).length) {
+      return <Alert variant="info" isInline isPlain title="This pipeline has no parameters." />;
+    }
+
+    return Object.entries(runParams).map(([label, value]) => {
+      const inputDefinitionParams = getInputDefinitionParams(version);
+      const parameterType = inputDefinitionParams?.[label].parameterType;
+      const inputProps = {
+        value,
+        id: label,
+        name: label,
+        onChange: (
+          _event: React.ChangeEvent<unknown> | null,
+          newValue: string | number | boolean,
+        ) => onChange({ ...runParams, [label]: newValue }),
+      };
+      let input: React.ReactNode;
+
+      switch (parameterType) {
+        case InputDefParamType.NumberInteger:
+          input = (
+            <NumberInputWrapper
+              {...inputProps}
+              min={0}
+              value={parseInt(String(value))}
+              onChange={(newValue) => inputProps.onChange(null, newValue)}
+            />
+          );
+          break;
+        case InputDefParamType.Boolean:
+          input = <RadioInputParam {...inputProps} value={String(value)} />;
```

Comment:
I might be missing something, but can you make the value a boolean. It looks like you convert it to a boolean on submit

Reply:
Switched to boolean.